redis/src/aio/connection_manager.rs

```diff
@@ -113,10 +127,27 @@ impl Default for ConnectionManagerConfig {
             response_timeout: Self::DEFAULT_RESPONSE_TIMEOUT,
             connection_timeout: Self::DEFAULT_CONNECTION_TIMEOUT,
             push_sender: None,
+            resubsrcibe_automatically: false,
         }
     }
 }
 
+struct Internals {
+    /// Information used for the connection. This is needed to be able to reconnect.
+    client: Client,
+    /// The connection future.
+    ///
+    /// The `ArcSwap` is required to be able to replace the connection
+    /// without making the `ConnectionManager` mutable.
+    connection: ArcSwap<SharedRedisFuture<MultiplexedConnection>>,
+
+    runtime: Runtime,
+    retry_strategy: ExponentialBuilder,
+    connection_config: AsyncConnectionConfig,
+    subscription_tracker: Option<Mutex<SubscriptionTracker>>,
```

Comment:
As far as I see it's also possible to use standard mutex, there is no need to hold guard across await points.



ref: https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use

Reply:
true, but I've learned that standard mutexes could stil cause a runtime to slow down, if they're stopping multiple tasks.

Reply:
Oh, I didn't know can you point me to a reference, if you can reach it easily ?

Reply:
(feel free to resolve the review)

Reply:
it's a very simple issue - instead of yielding to another task, the runtime's thread is blocked on the mutex. , thus blocking all tasks on that thread. while I believe this isn't an issue here, since the mutex would be held very shortly, I prefer not to optimize in a way that is potentially harmful to the users.